library/nametab.ml

```diff
@@ -212,6 +210,36 @@ let push visibility uname o tab =
     Id.Map.add id ptab tab
 
 
+let rec remove_path uname tree = function
+  | modid :: path ->
+      let map =
+        try
+          let submap = ModIdmap.find modid tree.map in
+          let submap = remove_path uname submap path in
+          if is_empty_tree submap then ModIdmap.empty
+          else ModIdmap.add modid submap tree.map
+        with Not_found ->
+          (* The name was actually not here *)
+          tree.map
+      in
+      let this =
+        let test = function Relative (uname',_) -> not (U.equal uname uname') | _ -> true in
+        List.filter test tree.path
+      in
+      mktree this map
+  | [] ->
+      let this =
+        let test = function Absolute (uname',_) -> not (U.equal uname uname') | _ -> true in
+        List.filter test tree.path
+      in
+      mktree this tree.map
+
+let remove uname tab =
```

Comment:
@herbelin can we document what this function does?

Reply:
I added a comment of this function in #16945.